### PR TITLE
Added GitHub Actions workflow file and support for new and old (down to 1.8) versions of SM

### PR DESCRIPTION
**Pull request by shqke**
- Replaced ke::AString, ke::AutoPtr, ke::Vector (don't exist in newer SM) with STL alternatives.
- Enforced `-std=c++14` (std::make_unique).
- Enforced `_GLIBCXX_USE_CXX11_ABI=0` - `std::string::replace` requires glibc `3.4.21` on upgraded debian 10.
- Support for clang 10 (packaged with ubuntu 20).
- Added local smsdk_ext files to support older SM versions.

Didn't touch anything else. Only tested on Left 4 Dead 2 under debian 10.

### CODE REVIEW

File: smsdk_config.h [hunk 1]
```diff
@@ -40,7 +40,7 @@
 /* Basic information exposed publicly */
 #define SMEXT_CONF_NAME			"SourceTV Manager"
 #define SMEXT_CONF_DESCRIPTION	"Interface to interact with the SourceTV server."
-#define SMEXT_CONF_VERSION		"1.2"
+#define SMEXT_CONF_VERSION		"1.3"
```
Comment (peace-maker) [inline smsdk_config.h:43]: No need to bump the version since 1.2 wasn't released yet.

**Review by peace-maker — CHANGES_REQUESTED** [review]
Thank you! This looks awesome and will help keep this extension alive! A few comments inline.

Please remove the `smsdk_ext.cpp/h` files again, there's no point supporting that old sourcemod builds and they're in sourcemod's public folder since SM 1.6.

I'm happy to take this as is and change it up myself, if you don't want to modify it again :)

**Comment by shqke** [comment]
Removal of smsdk_ext files will get rid of SM1.9 support that is still quite common in L4D community, which is mostly taken by the lack of upgrades on some old and yet popular server extensions or plugins.

Local smsdk_ext/h are there simply because of META_IFACE_OK in [smsdk_ext.cpp](https://github.com/alliedmodders/sourcemod/blob/1.9-dev/public/smsdk_ext.cpp#L320), definition of which in [ISmmPlugin.h](https://github.com/alliedmodders/metamod-source/blob/1.10-dev/core/ISmmPlugin.h#L49) is only present in 1.10.

It's all but these two recent files, and they've had a support for in a builder script already, so I've just added them to reduce hassle for server owners.

**Comment by peace-maker** [comment]
Wait, I don't get it. I can compile using sourcemod 1.9 just fine. So the problem is with compiling against older metamod-source versions than 1.10? How does including recent smdk_ext.cpp/h files help here? And who is running metamod-source 1.9?

**Comment by shqke** [comment]
> How does including recent smdk_ext.cpp/h files help here?

My bad, not recent but older that still uses IFACE_* constants from sdk.

On a breaking change, can ditch them and 1.9 (and lower) again as if nothing.

**Comment by peace-maker** [comment]
Can you tell me a sourcemod and metamod version combination that you want to support, that requires this file? I don't like having files from sourcemod'd sdk in tree, potentially getting out of sync in the future, for some ancient builds trying to compile against metamod source 1.9 from 2013 :(

You can compile this extension against sourcemod 1.9 and lower just fine using metamod 1.10.

**Comment by shqke** [comment]
Anything else?

**Comment by peace-maker** [comment]
Perfect, thank you a ton!

**Pull request merged**